aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -197,7 +198,28 @@ int64_t hsum(const uint8_t* A, int len) {
   for (const auto k : c10::irange(8)) {
     row_sum += temp[k];
   }
-#endif // CPU_CAPABILITY_AVX2
+#elif defined(CPU_CAPABILITY_AVX512)
+  __m512i sum_v = _mm512_setzero_si512();
+  __m512i one_epi16_v = _mm512_set1_epi16(1);
+  __m512i one_epi8_v = _mm512_set1_epi8(1);
+  // vectorized
+  for (; i < len / 64 * 64; i += 64) {
+    __m512i src_v = _mm512_loadu_si512(reinterpret_cast<__m512i const*>(A + i));
+    sum_v = _mm512_add_epi32(
+      sum_v,
+      _mm512_madd_epi16(
+        // first argument is unsigned, second is signed
+        _mm512_maddubs_epi16(src_v, one_epi8_v),
+      one_epi16_v)
+    );
```

Comment:
Wondering why we couldn't have used the Vec class here. Maybe a good refactor opportunity later

Reply:
This one's actually not present in `vec512_qint.h`, and neither is its counterpart present in `vec256_qint.h`. But I agree that moving them to those files seems to be a good idea.

Reply:
@ezyang, I'm also okay with refactoring the code in this PR.
Or I can create a follow-up PR so that I wouldn't lose track of refactoring opportunities like this one.

Reply:
let's do it in a follow up